### ISSUE TITLE
ArrayIndexOutOfBoundsException on parsing

### ISSUE THREAD

**Issue by lycis** [HubSpot/jinjava#34]
Hi, 

I am parsing the following jinja template:

```jinja
{% import 'library.jinja' as lib %}
{{ lib.echo('test') }}
```

`library.jinja` is defined as this:

```jinja
{% macro echo(what) -%}
echo {{ what }}
{%- endmacro %}
```

For some reason that I can not figure out I am ending with the following Exeception when calling
```java
JinjavaConfig.Builder configBuilder = JinjavaConfig.newBuilder().withLstripBlocks(true).withTrimBlocks(true);
Jinjava jinjava = new Jinjava(configBuilder.build());

String template = "{% import 'library.jinja' as lib %}\n{{ lib.echo('test') }}";
String renderedTemplate = jinjava.render(template, null);
```` 

Exception:
```
java.lang.ArrayIndexOutOfBoundsException: 55
	at com.hubspot.jinjava.tree.parse.TokenScanner.newToken(TokenScanner.java:247)
	at com.hubspot.jinjava.tree.parse.TokenScanner.getNextToken(TokenScanner.java:172)
	at com.hubspot.jinjava.tree.parse.TokenScanner.computeNext(TokenScanner.java:281)
	at com.hubspot.jinjava.tree.parse.TokenScanner.computeNext(TokenScanner.java:31)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
	at com.google.common.collect.Iterators$PeekingImpl.hasNext(Iterators.java:1144)
	at com.hubspot.jinjava.tree.TreeParser.text(TreeParser.java:101)
	at com.hubspot.jinjava.tree.TreeParser.nextNode(TreeParser.java:81)
	at com.hubspot.jinjava.tree.TreeParser.buildTree(TreeParser.java:59)
	at com.hubspot.jinjava.interpret.JinjavaInterpreter.parse(JinjavaInterpreter.java:116)
	at com.hubspot.jinjava.lib.tag.ImportTag.interpret(ImportTag.java:86)
	at com.hubspot.jinjava.lib.tag.Tag.interpretOutput(Tag.java:29)
	at com.hubspot.jinjava.tree.TagNode.render(TagNode.java:50)
	at com.hubspot.jinjava.interpret.JinjavaInterpreter.render(JinjavaInterpreter.java:182)
	at com.hubspot.jinjava.interpret.JinjavaInterpreter.render(JinjavaInterpreter.java:155)
	at com.hubspot.jinjava.Jinjava.renderForResult(Jinjava.java:204)
	at com.hubspot.jinjava.Jinjava.renderForResult(Jinjava.java:177)
	at com.hubspot.jinjava.Jinjava.render(Jinjava.java:150)
```

I do not know if this is a bug in Jinjava or in my code. But I can not figure out what I would be doing wrong.

I just figured out that I do not experience this error when I do not use
```java
JinjavaConfig.newBuilder().withLstripBlocks(true).withTrimBlocks(true);
```
Just using the following works out fine:
```java
JinjavaConfig.Builder configBuilder = JinjavaConfig.newBuilder().build();
```

**Comment by jaredstehler** [comment]
which version of jinjava are you using?

**Comment by lycis** [comment]
I am using maven to pull this one:
```
<dependency>
  <groupId>com.hubspot.jinjava</groupId>
  <artifactId>jinjava</artifactId>
  <version>2.1.0</version>
</dependency>
```

**Comment by jaredstehler** [comment]
cool, I am able to reproduce with the latest code, and there does appear to be an issue with the scanner w.r.t. lstrip and/or trim blocks. I'll investigate.

**Comment by jaredstehler** [comment]
fixed in `2.1.4-SNAPSHOT` via da01d87

**Comment by jaredstehler** [comment]
I'll release 2.1.4 shortly.

**Comment by jaredstehler** [comment]
I've released `2.1.4`; should be live in maven central in 4 hours or so.
